tests/socket-read-type.phpt

```diff
@@ -0,0 +1,63 @@
+--TEST--
+Test of Socket::read() - testing type \Socket::PHP_NORMAL_READ and \Socket::PHP_BINARY_READ
+--FILE--
+<?php
+	/* Setup socket server */
+    $server = new Socket(\Socket::AF_INET, \Socket::SOCK_STREAM, getprotobyname('tcp'));
+    if (!$server) {
+        die('Unable to create AF_INET socket [server]');
+    }
+    $bound = false;
+    for($port = 31337; $port < 31357; ++$port) {
+        if ($server->bind('127.0.0.1', $port)) {
```

Comment:
This does not always work for me. If it attempts to bind to an already bound port, then I get a `RuntimeException`. It needs to be placed in a try...catch block:
```php
try {
    if ($server->bind('127.0.0.1', $port)) {
        $bound = true;
        break;
    }
} catch (RuntimeException $re) {}
```